### PR TITLE
Create a database based on a diagram

### PR DESCRIPTION
**Pull request by OybekKayumov**
### In this pull request tasks are done:
- Create new tables.
- Convert diagrams and arrows connections into tables.
- Use Many to Many relationships.

## File added
`schema_based_on_diagram.sql`

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  FOREIGN KEY (patient_id) REFERENCES patients(id);
+
+CREATE TABLE histories_treatment(
+	id INT GENERATED ALWAYS AS IDENTITY,
+	treatment_id INT,
+	medical_history_id INT,
+	CONSTRAINT fk_treatment_id FOREIGN KEY (treatment_id) REFERENCES treatments(id),
+	CONSTRAINT fk_medical_history_id FOREIGN KEY (medical_history_id) REFERENCES medical_histories(id),
+	PRIMARY KEY(id, treatment_id, medical_history_id)
+);
```
Comment (gemmen29) [inline schema_based_on_diagram.sql:61]: - [ ] Please add indexes for all foreign keys as in the requirement in order to make the search with foreign keys faster.
image

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 💯 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	FOREIGN KEY (medical_history_id) REFERENCES medical_histories(id);
+
+ALTER TABLE invoice_items ADD CONSTRAINT fk_invoice_id 
+  FOREIGN KEY (invoices_id) REFERENCES invoices(id);
+
+ALTER TABLE invoice_items ADD CONSTRAINT fk_treatment_id 
+  FOREIGN KEY (treatment_id) REFERENCES treatments(id);
+
+ALTER TABLE medical_histories ADD CONSTRAINT fk_patient_id 
+  FOREIGN KEY (patient_id) REFERENCES patients(id);
```
Comment (ShadyShawkat) [inline schema_based_on_diagram.sql:52]: - [ ] Kindly add the indices for the rest of the foreign keys, as suggested by the previous reviewer.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi Team, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

Image

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql (hunk 2 above)
Comment (OybekKayumov) [reply schema_based_on_diagram.sql:52]: Thank you for your feedback. All indexes were created.

File: schema_based_on_diagram.sql [hunk 3]
```diff
@@ -60,7 +60,10 @@ CREATE TABLE histories_treatment(
 	PRIMARY KEY(id, treatment_id, medical_history_id)
 );
 
+CREATE INDEX invoices_medical_history_id_idx ON invoices(medical_history_id);
+CREATE INDEX invoice_items_invoice_id_idx ON invoice_items(invoices_id);
+CREATE INDEX invoice_items_treatment_id_idx ON invoice_items(treatment_id);
+CREATE INDEX medical_histories_patient_id_idx ON medical_histories(patient_id);
```
Comment (Cathella) [inline schema_based_on_diagram.sql:66]: ✅ Added indexes 👍🏻

**Review by Cathella — APPROVED** [review]
## 🔥 Status: Approved 🟢

### Hi @OybekKayumov, @keza681 👋🏻, 

Good job on making the requested changes 💪🏻!
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

rosycheeks-mochi-peach


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**